### PR TITLE
Added sample interface (by Zhenyu),and refactor all interfaces for rD…

### PR DESCRIPTION
**Pull request by mcfatealan**
…SN.Monitor

### CODE REVIEW

File: src/core/tools/common/profiler_command.cpp [hunk 1]
```diff
@@ -252,52 +252,291 @@ namespace dsn {
...
+
+                    if ((i == TASK_CODE_INVALID) || (s_spec_profilers[task_id].is_profile == false))
+                        continue;
+                    ss << "\""<<dsn_task_code_to_string(task_id) << "\",";
+
+                }
+                ss << "]";
+                return ss.str();
+            }
+            //return a list of 2 elements for a speciafic task:
```
Comment (imzhenyu) [inline src/core/tools/common/profiler_command.cpp]: specific?

File: src/core/tools/common/profiler_command.cpp [hunk 2]
```diff
@@ -252,52 +252,291 @@ namespace dsn {
...
-            int task_id;
-            perf_counter_ptr_type counter_type;
-            counter_percentile_type percentile_type;
             std::stringstream ss;
 
-            for (int i = 0; i <= dsn_task_code_max(); ++i)
+            if (args.size() < 1)
             {
-                task_id = i;
+                ss << "unenough arguments" << std::endl;
```
Comment (imzhenyu) [inline src/core/tools/common/profiler_command.cpp]: incorrect arguments

File: src/core/tools/common/profiler_command.cpp [hunk 3]
```diff
@@ -252,52 +252,291 @@ namespace dsn {
...
+                {
+                    ss << "unenough arguments" << std::endl;
+                    return ss.str();
+                }
+
+                int task_id = find_task_id(args[1]);
+
+                if ((task_id == TASK_CODE_INVALID) || (s_spec_profilers[task_id].is_profile == false))
+                {
+                    ss << "no such task code" << std::endl;
```
Comment (imzhenyu) [inline src/core/tools/common/profiler_command.cpp]: or target task is not profiled (is_profile = false)

File: src/core/tools/common/profiler_command.cpp (hunk 1 above)
Comment (mcfatealan) [inline src/core/tools/common/profiler_command.cpp]: no, it returns a list like: ["RPC_NFS_COPY","RPC_PREPARE"...]

File: src/core/tools/common/profiler_command.cpp (hunk 3 above)
Comment (mcfatealan) [inline src/core/tools/common/profiler_command.cpp]: Thanks, I'll fix it tomorrow.

File: src/core/tools/common/profiler_command.cpp (hunk 1 above)
Comment (imzhenyu) [inline src/core/tools/common/profiler_command.cpp]: I mean the word "speciafic" is not spelled correctly:-)

Comment (mcfatealan) [inline src/core/tools/common/profiler_command.cpp]: ....sorry

**Pull request merged**